### DISCUSSION TITLE
When will Codespaces be out of beta?

### DISCUSSION THREAD

**Discussion by lloydjatkinson** [community/community#3665] [Codespaces] [answered]
Hi,

I remember February being the original goal but it's now passed. I was using Visual Studio Online but then when it was killed off, I signed up for the "migrate Visual Studio Online users to Codespaces" thing but that never happened after the follow up emails for some reason.

I've not been able to use either since then. It's quite frustrating, if users can't be migrated it shouldn't have been offered, and codespaces still isn't available.

Is there any ETA at all?

**Comment by PranamBhat** [comment]
Hi @lloydjatkinson 

You can always share your feedback here [https://support.github.com/contact/feedback?category=other&subject=GitHub+public+roadmap](https://support.github.com/contact/feedback?category=other&subject=GitHub+public+roadmap)


Also, check out the GitHub Roadmap : [https://github.com/github/roadmap/projects/1](https://github.com/github/roadmap/projects/1)

**Comment by 2percentsilk** [comment, marked as answer]
Hi @lloydjatkinson, 

I'm sorry to hear this frustration. Our goal was to make that transition as easy and smooth as possible for our VSO users but clearly we missed the mark here. I'm looking into our records for your submission and will ensure you're migrated or I will reach out if we need more details to get you moved over.

In the meantime, Codespaces is set to GA sometime between April - June, according to our [roadmap](https://github.com/github/roadmap/issues/55).
